src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -23,11 +23,25 @@ class FirewallContext
 {
     private $listeners;
     private $exceptionListener;
+    private $config;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null)
+    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
     {
         $this->listeners = $listeners;
         $this->exceptionListener = $exceptionListener;
+
+        if (null === $config) {
+            @trigger_error(sprintf('"%s()" expects an instance of "%s" as third argument since version 3.2 and will throw an exception in 4.0 if not provided.', __METHOD__, FirewallConfig::class), E_USER_DEPRECATED);
```

Comment:
@HeahDude Sorry, I got it now.

There is no "good" reason to break BC here. But in the other hand, I can't find a good one to have `FirewallContext::$config` being null, as we shouldn't be able to get an instance of `FirewallContext` without configuring a firewall (it is already documented as "a wrapper around the current firewall configuration" 😄 ).

Also as you pointed out, this deprecation is not mandatory, I stay ready to remove it if your point comes to be shared.